cellpack/autopack/Environment.py

```diff
@@ -280,6 +280,23 @@ def _setup(self):
                 self.referenced_objects,
             ) = Recipe.resolve_composition(self.recipe_data)
             self.create_objects()
+        if self.use_gradient:
+            gradients = self.recipe_data["gradients"]
```

Comment:
Only mildly redundant to have a boolean `use_gradient` here along with the presence or absence of a gradient in `self.recipe_data["gradients"]`. Could we collapse this check into a check for existence in recipe_data?